tests/core/util/test_file_keyring_synchronization.py

```diff
@@ -30,17 +31,13 @@ def dummy_set_passphrase(service, user, passphrase, keyring_path, index, num_wor
         with open(ready_file_path, "w") as f:
             f.write(f"{os.getpid()}\n")
 
-        # Wait up to 30 seconds for all processes to indicate readiness
+        # Wait up to 120 seconds for all processes to indicate readiness
         start_file_path: Path = Path(ready_file_path.parent) / "start"
-        remaining_attempts = 120
-        while remaining_attempts > 0:
-            if start_file_path.exists():
-                break
-            else:
-                sleep(0.25)
-                remaining_attempts -= 1
+        start = time.time()
+        while not start_file_path.exists() and time.time() - start < 120:
+            sleep(0.1)
 
-        assert remaining_attempts >= 0
+        assert start_file_path.exists()
```

Comment:
Whats the exact scenario you are worried about here?

Reply:
Questionable practices with race conditions in them floating around our code leading to more people following those patterns because we already do that thing.  Also, every future reader of the code having to debate with themselves whether this race risk is relevant or not.

Reply:
See 499e15074e1df422034e3efc846f1e1cff911f10 + turned out i refactored away one of the checks before, see 3440ad3589e5d6be09ee445dd9574d0134936e5b.